Assets/Scripts/Singletons/DataModelSingletons/PlayerManager.cs

```diff
@@ -130,6 +135,17 @@ public static void SetPlayerName(string playerName)
         Debug.Log($"Player name set to: {Instance.PlayerName}");
     }
 
+    private void updatePlayerFacingDirectionVector(Vector3 directionVector)
+    {
+        // If the player is not moving then then Movement Vector in PlayerMovement is 0
+        // As we want a direction we only take the last non 0 Movement Vector
+        if (directionVector != Vector3.zero)
+        {
+            PlayerFacingDirection = new Vector3(directionVector.x, 0f, directionVector.y);

```

Comment:
Is this meant to be ```(direction.x, 0f, direction.z)```?

Reply:
Also, if we are sending the MovementVector from the PlayerMovement to the singleton, and setting the y axis element to 0, why can't we just the MovementVector variable instead of this argument?

Reply:
If we set it to (direction.x, 0f, direction.z) it won't work. I think the vectors are not oriented in the same way.

Reply:
Ah yeah I forgot it's ```PlayerMovement.MovementVector``` that you were passing in. And that is the virtual axis, which is basically a cartesian system. So this works fine.